Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -295,6 +295,24 @@ extension NIOSSLCertificate {
         return try readCertificatesFromBIO(bio)
     }
 
+    /// Returns the timestamp before which this certificate is not valid.
+    ///
+    /// The value is in seconds since the UNIX epoch.
+    public var notValidBefore: Int {
```

Comment:
Should this be `NIODeadline`?

Reply:
I don't think so: `NIODeadline` refers to an instant in time but it is not associated with any other calendaring system. The backing storage is `uptimeNanoseconds`, and is a `UInt64`. This means any valid cert may have an unrepresentable `notBefore` time, as it will likely be before the system start time.

Reply:
@Lukasa the `uptimeNanoseconds` is a good point. Do we need another time type in NIO? Because I don't think a naked `Int` is the right type here...

Reply:
Swapped to time_t.